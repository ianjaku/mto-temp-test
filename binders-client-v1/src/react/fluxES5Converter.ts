/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
