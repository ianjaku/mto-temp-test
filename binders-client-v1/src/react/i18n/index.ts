import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
