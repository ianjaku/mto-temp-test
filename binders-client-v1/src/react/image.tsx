/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
