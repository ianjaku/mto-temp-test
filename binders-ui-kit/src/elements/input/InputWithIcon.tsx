import * as React from "react";
import ThemedInput, { IInputProps } from "./index";
import { omit } from "ramda";
import "./input.styl";

export interface IIconInputProps extends IInputProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    icon?: any;
}

const InputWithIcon: React.FC<Omit<IIconInputProps, "ref">> = (props) => {
    return (
        <div className="input-wrapper input-wrapper-icon">
            {props.icon && (
                <span className="input-icon">{props.icon}</span>
            )}
            <ThemedInput {...omit(["icon"], props)} />
        </div>
    );
}

export default InputWithIcon;
