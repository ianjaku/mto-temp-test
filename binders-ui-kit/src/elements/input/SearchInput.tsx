import * as React from "react";
import ThemedInput, { IInputProps } from "./index";
import SearchIcon from "../icons/Search";
import cx from "classnames";
import "./input.styl";

const SearchInput: React.FC<Omit<IInputProps, "ref">> = (props) => {
    return (
        <div className={cx("input-wrapper input-wrapper-search", { "input-wrapper--inverted": props.inverted })}>
            <span className="input-icon input-icon--center">
                <SearchIcon />
            </span>
            <ThemedInput {...props} type="text" />
        </div>
    );
}

export default SearchInput;
