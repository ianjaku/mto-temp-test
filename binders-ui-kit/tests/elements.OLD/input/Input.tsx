import * as React from "react";
import AccountIcon from "../../../src/elements/icons/Account";
import Input from "../../../src/elements/input";
import InputWithIcon from "../../../src/elements/input/InputWithIcon";
import SearchInput from "../../../src/elements/input/SearchInput";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const CONTENT = "Search for documents and collections";
const EMAIL = "john.doe@manual.to";
const PLACEHOLDER = "your name";

const createEmptyTextInput = () => wrapWithTheme(<Input type="text" />);
const createContentPasswordInput = () => wrapWithTheme(<Input type="password" value={CONTENT} />);
const createContentIconInput = () => wrapWithTheme(<InputWithIcon type="email" value={EMAIL} icon={<AccountIcon />} />);
const createSearchInput = () => wrapWithTheme(<SearchInput name="search" />);
const createDisabledInput = () => wrapWithTheme(<Input disabled={true} name="full_name" />);
const createInputWithPlaceholder = () => wrapWithTheme(<Input placeholder={PLACEHOLDER} type="text" name="full_name" />);
const createInvertedSearchInput = () => wrapWithTheme(<SearchInput name="search" inverted={true} />);

// most preferred
describe("Input", () => {

    test("Empty text input (snapshot)", () => {
        const input = create(createEmptyTextInput()).toJSON();
        expect(input).toMatchSnapshot();
    });

    test("Content password input (snapshot)", () => {
        const input = create(createContentPasswordInput()).toJSON();
        expect(input).toMatchSnapshot();
    });

    test("Content icon input (snapshot)", () => {
        const input = create(createContentIconInput()).toJSON();
        expect(input).toMatchSnapshot();
    });

    test("Search input (snapshot)", () => {
        const input = create(createSearchInput()).toJSON();
        expect(input).toMatchSnapshot();
    });

    test("Disabled text input (snapshot)", () => {
        const input = create(createDisabledInput()).toJSON();
        expect(input).toMatchSnapshot();
    });

    test("Input with placeholder (snapshot)", () => {
        const input = create(createInputWithPlaceholder()).toJSON();
        expect(input).toMatchSnapshot();
    });

    test("Inverted search input (snapshot)", () => {
        const input = create(createInvertedSearchInput()).toJSON();
        expect(input).toMatchSnapshot();
    });

    test("Input should be of type text and be empty (mount: enzyme+jsdom)", () => {
        const element = mount(createEmptyTextInput())
            .find(".input")
            .at(0);
        expect(element.exists()).toEqual(true);
        expect(element.prop("type")).toEqual("text");
        expect(element.text()).toEqual("");
    });

    test("Password input should contain a value and be of type password (mount: enzyme+jsdom)", () => {
        const element = mount(createContentPasswordInput())
            .find(".input")
            .at(0);
        expect(element.exists()).toEqual(true);
        expect(element.prop("type")).toEqual("password");
        expect(element.prop("value")).toEqual(CONTENT);
    });

    test("Content icon input should contain an account icon and in this case be of type email (mount: enzyme+jsdom)", () => {
        const element = mount(createContentIconInput())
            .find(".input")
            .at(0);
        expect(element.exists()).toEqual(true);
        expect(element.prop("type")).toEqual("email");
        expect(element.prop("value")).toEqual(EMAIL);

        const icon = element.find(".input-icon");
        expect(icon.exists());
    });


    test("Search input should have a search icon and here name search (mount: enzyme+jsdom)", () => {
        const element = mount(createSearchInput())
            .find(".input-wrapper-search .input");
        expect(element.exists()).toEqual(true);
        expect(element.prop("type")).toEqual("text");
        expect(element.prop("name")).toEqual("search");

        const icon = element.find(".input-icon");
        expect(icon.exists());
    });

    test("Disabled textbox should be disabled (mount: enzyme+jsdom)", () => {
        const element = mount(createDisabledInput())
            .find(".input")
            .at(0);
        expect(element.prop("disabled")).toEqual(true);
        expect(element.prop("name")).toEqual("full_name");
    });

    test("Placeholder should work (mount: enzyme+jsdom)", () => {
        const element = mount(createInputWithPlaceholder())
            .find(".input")
            .at(0);
        expect(element.prop("placeholder")).toEqual(PLACEHOLDER);
        expect(element.prop("name")).toEqual("full_name");
        expect(element.prop("type")).toEqual("text");
    });

    test("Inverted search input (mount: enzyme+jsdom)", () => {
        const element = mount(createInvertedSearchInput())
            .find(".input")
            .at(0);
        expect(element.prop("className")).toEqual("input input--inverted");
    });
});
