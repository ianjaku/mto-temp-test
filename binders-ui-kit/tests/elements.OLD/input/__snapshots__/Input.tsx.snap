// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Content icon input (snapshot) 1`] = `
<div
  className="input-wrapper input-wrapper-icon"
>
  <span
    className="input-icon"
  >
    <span
      className="material-icons "
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": "20px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      person
    </span>
  </span>
  <input
    className="input"
    onChange={[Function]}
    style={
      Object {
        "width": "undefinedpx",
      }
    }
    type="email"
    value="john.doe@manual.to"
  />
</div>
`;

exports[`Input Content password input (snapshot) 1`] = `
<input
  className="input"
  onChange={[Function]}
  style={
    Object {
      "width": "undefinedpx",
    }
  }
  type="password"
  value="Search for documents and collections"
/>
`;

exports[`Input Disabled text input (snapshot) 1`] = `
<input
  className="input"
  disabled={true}
  name="full_name"
  onChange={[Function]}
  style={
    Object {
      "width": "undefinedpx",
    }
  }
/>
`;

exports[`Input Empty text input (snapshot) 1`] = `
<input
  className="input"
  onChange={[Function]}
  style={
    Object {
      "width": "undefinedpx",
    }
  }
  type="text"
/>
`;

exports[`Input Input with placeholder (snapshot) 1`] = `
<input
  className="input"
  name="full_name"
  onChange={[Function]}
  placeholder="your name"
  style={
    Object {
      "width": "undefinedpx",
    }
  }
  type="text"
/>
`;

exports[`Input Inverted search input (snapshot) 1`] = `
<div
  className="input-wrapper input-wrapper-search input-wrapper--inverted"
>
  <span
    className="input-icon"
  >
    <span
      className="material-icons "
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      search
    </span>
  </span>
  <input
    className="input input--inverted"
    name="search"
    onChange={[Function]}
    style={
      Object {
        "width": "undefinedpx",
      }
    }
    type="text"
  />
</div>
`;

exports[`Input Search input (snapshot) 1`] = `
<div
  className="input-wrapper input-wrapper-search"
>
  <span
    className="input-icon"
  >
    <span
      className="material-icons "
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      search
    </span>
  </span>
  <input
    className="input"
    name="search"
    onChange={[Function]}
    style={
      Object {
        "width": "undefinedpx",
      }
    }
    type="text"
  />
</div>
`;
